gemini/sample-apps/llamaindex-rag/backend/indexing/vector_search_utils.py

```diff
@@ -0,0 +1,142 @@
+        for endpoint in aiplatform.MatchingEngineIndexEndpoint.list(
+            filter=f"display_name={index_endpoint_name}"
+        )
+    ]
+
+    if len(endpoint_names) == 0:
+        print(f"Creating Vector Search index endpoint {index_endpoint_name} ...")
+        vs_endpoint = aiplatform.MatchingEngineIndexEndpoint.create(
+            display_name=index_endpoint_name, public_endpoint_enabled=True
+        )
+        print(
+            f"Vector Search index endpoint {vs_endpoint.display_name} \
+                created with resource name {vs_endpoint.resource_name}"
+        )
+    else:
+        vs_endpoint = aiplatform.MatchingEngineIndexEndpoint(
+            index_endpoint_name=endpoint_names[0]
+        )
+        print(
+            f"Vector Search index endpoint {vs_endpoint.display_name} \
+                exists with resource name {vs_endpoint.resource_name}"
+        )
+    return vs_endpoint
+
+
+def deploy_index(
+    vs_index: aiplatform.MatchingEngineIndex,
+    vs_endpoint: aiplatform.MatchingEngineIndexEndpoint,
+    vector_index_name: str,
+):
+    index_endpoints = [
+        (deployed_index.index_endpoint, deployed_index.deployed_index_id)
+        for deployed_index in vs_index.deployed_indexes
+    ]
+
+    if len(index_endpoints) == 0:
+        print(
+            f"Deploying Vector Search index {vs_index.display_name} \
+                at endpoint {vs_endpoint.display_name} ..."
+        )
+        vs_deployed_index = vs_endpoint.deploy_index(
+            index=vs_index,
+            deployed_index_id=vector_index_name,
+            display_name=vector_index_name,
+            machine_type="e2-standard-16",
+            min_replica_count=1,
+            max_replica_count=1,
+        )
+        print(
+            f"Vector Search index {vs_index.display_name} \
+                is deployed at endpoint {vs_deployed_index.display_name}"
+        )
+    else:
+        vs_deployed_index = aiplatform.MatchingEngineIndexEndpoint(
+            index_endpoint_name=index_endpoints[0][0]
+        )
+        print(
+            f"Vector Search index {vs_index.display_name} \
+                is already deployed at endpoint {vs_deployed_index.display_name}"
+        )
+    return vs_deployed_index
+
+
+def get_existing_index_and_endpoint(vector_index_name: str, index_endpoint_name: str):
+    # Check for existing index
+    index = None
+    index_list = aiplatform.MatchingEngineIndex.list(
+        filter=f"display_name={vector_index_name}"
+    )
+    if index_list:
+        index = index_list[0]
+        print(f"Found existing index: {index.display_name}")
+
+    # Check for existing endpoint
+    endpoint = None
+    endpoint_list = aiplatform.MatchingEngineIndexEndpoint.list(
+        filter=f"display_name={index_endpoint_name}"
+    )
+    if endpoint_list:
+        endpoint = endpoint_list[0]
+        print(f"Found existing endpoint: {endpoint.display_name}")
+
+    return index, endpoint
```

Comment:
Is there a reason for this method when the other "create" methods also check for a previously created index/endpoint?

Reply:
Removed redundant creation in latest commit